### PR TITLE
#77 Create handler for "team" command

### PR DESCRIPTION
**Pull request by petrokramar**
Connect to #77

### CODE REVIEW

File: src/main/java/juja/microservices/gamification/slackbot/dao/impl/RestGamificationRepository.java [hunk 1]
```diff
@@ -132,4 +131,29 @@ public RestGamificationRepository(RestTemplate restTemplate) {
...
+        try {
+            logger.debug("Started request to Gamification service. Request is : [{}]", request.toString());
+            ResponseEntity<String[]> response = restTemplate.exchange(gamificationSendTeamUrl,
+                    HttpMethod.POST, request, String[].class);
+            result = response.getBody();
+            logger.debug("Finished request to Gamification service. Response is: [{}]", response.toString());
+        } catch (HttpClientErrorException ex) {
+            ApiError error = Utils.convertToApiError(ex);
+            logger.warn("Gamification service returned an error: [{}]", error);
+            throw new GamificationExchangeException(Utils.convertToApiError(ex), ex);
```
Comment (OlgaKulikova) [inline src/main/java/juja/microservices/gamification/slackbot/dao/impl/RestGamificationRepository.java]: `Utils.convertToApiError(ex)`  is already in the variable `error`. Use it.

File: src/main/java/juja/microservices/gamification/slackbot/dao/impl/RestGamificationRepository.java [hunk 2]
```diff
@@ -132,4 +131,29 @@ public RestGamificationRepository(RestTemplate restTemplate) {
...
+        }
+
+        logger.info("Saved Team achievements: [{}]", Arrays.toString(result));
+        return result;
+    }
+
+
```
Comment (OlgaKulikova) [inline src/main/java/juja/microservices/gamification/slackbot/dao/impl/RestGamificationRepository.java]: please delete extra empty lines

File: src/main/java/juja/microservices/gamification/slackbot/exceptions/TeamExchangeException.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package juja.microservices.gamification.slackbot.exceptions;
+
+public class TeamExchangeException extends BaseBotException {
+    public TeamExchangeException(ApiError error, Exception ex) {
+        super(error, ex);
+    }
+
+
```
Comment (OlgaKulikova) [inline src/main/java/juja/microservices/gamification/slackbot/exceptions/TeamExchangeException.java]: Please delete extra empty lines

File: src/main/java/juja/microservices/gamification/slackbot/service/impl/DefaultGamificationService.java [hunk 4]
```diff
@@ -110,6 +117,35 @@ public String sendInterviewAchievement(String fromUser, String text) {
...
+
+        logger.debug("Start create Team achievement from slack parsed command");
+        String fromUuid = createSlackParsedCommand(fromUser, text).getFromUser().getUuid();
+        TeamDTO teamDTO = teamService.getTeamByUserUuid(fromUuid);
+        int teamSize = teamDTO.getMembers().size();
+
+        TeamAchievement team = new TeamAchievement(fromUuid, teamDTO.getMembers());
+        logger.debug("Team achievement was created. team: {}", team.toString());
+
+        Set<UserDTO> users = userService.findUsersByUuids(teamDTO.getMembers());
```
Comment (OlgaKulikova) [inline src/main/java/juja/microservices/gamification/slackbot/service/impl/DefaultGamificationService.java]: `teamDTO.getMembers()` you invoke several times, use local variable for that.

File: src/main/java/juja/microservices/gamification/slackbot/service/impl/DefaultGamificationService.java [hunk 5]
```diff
@@ -110,6 +117,35 @@ public String sendInterviewAchievement(String fromUser, String text) {
...
 
+    @Override
+    public String sendTeamAchievement(String fromUser, String text) {
+
+        logger.debug("Start sending and saving Team achievement. fromUser: [{}] text: [{}]", fromUser, text);
+
+        logger.debug("Start create Team achievement from slack parsed command");
+        String fromUuid = createSlackParsedCommand(fromUser, text).getFromUser().getUuid();
+        TeamDTO teamDTO = teamService.getTeamByUserUuid(fromUuid);
+        int teamSize = teamDTO.getMembers().size();
```
Comment (OlgaKulikova) [inline src/main/java/juja/microservices/gamification/slackbot/service/impl/DefaultGamificationService.java]: please create local variable closer to the place where you are going to use it.

File: src/test/java/juja/microservices/integration/GamificationSlackBotIntegrationTest.java [hunk 6]
```diff
@@ -260,6 +268,43 @@ public void returnErrorMessageIfThanksCommandConsistTwoOrMoreSlackNames() throws
                 .andExpect(content().string(INSTANT_MESSAGE));
     }
 
+//    @Test
```
Comment (OlgaKulikova) [inline src/test/java/juja/microservices/integration/GamificationSlackBotIntegrationTest.java]: if this test doesn't work please delete it

**Review by OlgaKulikova — CHANGES_REQUESTED** [review]

**Review by NikolayNN — APPROVED** [review]
it looks good

File: src/main/java/juja/microservices/gamification/slackbot/service/impl/DefaultGamificationService.java [hunk 7]
```diff
@@ -125,19 +125,20 @@ public String sendTeamAchievement(String fromUser, String text) {
...
-        Set<UserDTO> users = userService.findUsersByUuids(teamDTO.getMembers());
+        Set<UserDTO> users = userService.findUsersByUuids(teamMembers);
         Set<String> slackNames = new LinkedHashSet<>();
         users.forEach(user -> slackNames.add(user.getSlack()));
         logger.debug("Slack names for team {} were received: {}", team.toString(), slackNames);
 
         String[] ids = gamificationRepository.saveTeamAchievement(team);
         logger.info("Team achievement was saved with ids: {}", Arrays.toString(ids));
 
+        int teamSize = teamDTO.getMembers().size();
```
Comment (OlgaKulikova) [inline src/main/java/juja/microservices/gamification/slackbot/service/impl/DefaultGamificationService.java]: You already have teamMembers variable instead teamDTO.getMembers()

Comment (petrokramar) [reply src/main/java/juja/microservices/gamification/slackbot/service/impl/DefaultGamificationService.java]: Fixed

**Review by OlgaKulikova — APPROVED** [review]

**Pull request merged**